quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java

```diff
@@ -0,0 +1,137 @@
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
+        + "var array = new Uint8Array(5);"
+        + "array[0] = 86;"
+        + "array[1] = 7;"
+        + "array[2] = 53;"
+        + "array[3] = 0;"
+        + "array[4] = 9;"
+        + "array;");
+    assertThat(byteArray).asList()
+        .containsExactly((byte) 86, (byte) 7, (byte) 53, (byte) 0, (byte) 9);
+  }
+
+  @Test public void uint8ArraySliceConvertedToByteArray() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
+        + "var array = new Uint8Array(5);"
+        + "array[0] = 86;"
+        + "array[1] = 7;"
+        + "array[2] = 53;"
+        + "array[3] = 0;"
+        + "array[4] = 9;"
+        + "array.slice(1, 3);");
+    assertThat(byteArray).asList()
+        .containsExactly((byte) 7, (byte) 53);
+  }
+
+  @Test public void int8ArrayConvertedToByteArray() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
+        + "var array = new Int8Array(5);"
+        + "array[0] = 86;"
+        + "array[1] = 7;"
+        + "array[2] = 53;"
+        + "array[3] = 0;"
+        + "array[4] = 9;"
+        + "array;");
+    assertThat(byteArray).asList()
+        .containsExactly((byte) 86, (byte) 7, (byte) 53, (byte) 0, (byte) 9);
+  }
+
+  @Test public void uint8ClampedArrayConvertedToByteArray() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
+        + "var array = new Uint8ClampedArray(5);"
+        + "array[0] = 86;"
+        + "array[1] = 7;"
+        + "array[2] = 53;"
+        + "array[3] = 0;"
+        + "array[4] = 9;"
+        + "array;");
+    assertThat(byteArray).asList()
+        .containsExactly((byte) 86, (byte) 7, (byte) 53, (byte) 0, (byte) 9);
+  }
+
+  @Test public void int16ArrayUnsupported() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
+        + "var array = new Int16Array(1);"
+        + "array[0] = 8675;"
+        + "array;");
+    assertThat(byteArray).isNull();
+  }
+
+  @Test public void uint16ArrayUnsupported() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
+        + "var array = new Uint16Array(1);"
+        + "array[0] = 8675;"
+        + "array;");
+    assertThat(byteArray).isNull();
+  }
+
+  @Test public void int32ArrayUnsupported() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
+        + "var array = new Int32Array(1);"
+        + "array[0] = 8675;"
+        + "array;");
+    assertThat(byteArray).isNull();
+  }
+
+  @Test public void uint32ArrayUnsupported() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
+        + "var array = new Uint32Array(1);"
+        + "array[0] = 8675;"
+        + "array;");
+    assertThat(byteArray).isNull();
+  }
+
+  @Test public void float32ArrayUnsupported() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
+        + "var array = new Float32Array(1);"
+        + "array[0] = 867.5309;"
+        + "array;");
+    assertThat(byteArray).isNull();
+  }
+
+  @Test public void float64ArrayUnsupported() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
```

Comment:
`double[]`?

Reply:
Or maybe just leave as `Object` for now?

Reply:
Lemme use the actual type that makes sense.